src/ts/plugins/server/GitlabHostingConnector.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Silex website builder, free/libre no-code tool for makers.
+ * Copyright (c) 2023 lexoyo and Silex Labs foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Gitlab hosting connector
+ * @fileoverview Gitlab hosting connector for Silex, connect to the user's Gitlab account to host websites with .gitlab-ci.yml file template for plain html by default
+ * @see https://docs.gitlab.com/ee/api/oauth2.html
+ * @see https://docs.gitlab.com/ee/user/project/pages/getting_started/pages_ci_cd_template.html
+ */
+
+import GitlabConnector, { GitlabOptions, GitlabToken } from './GitlabConnector'
+import { HostingConnector, ConnectorFile } from '../../server/connectors/connectors'
+import { ConnectorType, WebsiteId, JobData, JobStatus, PublicationJobData } from '../../types'
+import { JobManager } from '../../server/jobs'
+import { join } from 'path'
+import { ServerConfig } from '../../server/config'
+
+//type GitlabSession = ConnectorSession
+//type GitlabOption = GitlabOptions
+type GitlabSession = Record<string, GitlabToken>
```

Comment:
this is already in GitlabConnector.ts, you probably want to export it from there and import it here

Reply:
done, ok thx